src/core/ext/filters/client_channel/lb_policy/xds/xds.cc

```diff
@@ -1178,23 +1139,23 @@ void XdsLb::ProcessChannelArgsLocked(const grpc_channel_args& args) {
   args_ = grpc_channel_args_copy_and_add_and_remove(
       &args, args_to_remove, GPR_ARRAY_SIZE(args_to_remove), &new_arg, 1);
   // Construct args for balancer channel.
-  grpc_channel_args* lb_channel_args =
-      BuildBalancerChannelArgs(*addresses, response_generator_.get(), &args);
-  // Create balancer channel if needed.
-  if (lb_channel_ == nullptr) {
-    char* uri_str;
-    gpr_asprintf(&uri_str, "fake:///%s", server_name_);
-    gpr_mu_lock(&lb_channel_mu_);
-    lb_channel_ = grpc_client_channel_factory_create_channel(
-        client_channel_factory(), uri_str,
-        GRPC_CLIENT_CHANNEL_TYPE_LOAD_BALANCING, lb_channel_args);
-    gpr_mu_unlock(&lb_channel_mu_);
-    GPR_ASSERT(lb_channel_ != nullptr);
-    gpr_free(uri_str);
+  grpc_channel_args* lb_channel_args = BuildBalancerChannelArgs(&args);
+  // Create or update balancer channel if needed.
+  if (balancer_name_changed_) {
```

Comment:
I think that if the balancer name changes, we should try to avoid having any period of time when we are not in contact with a balancer.  If we get the new name when the call to the old balancer is still functioning properly, then we should start up the new channel, start the balancer call on that new channel, and then wait for that call to be started and get its first serverlist before we swap out the old channel for the new channel.  However, if the call to the old balancer is not currently functioning properly (i.e., it is in retry delay), then we can swap them out immediately.